### DISCUSSION TITLE
H5easy Eigen vector shape question

### DISCUSSION THREAD

**Discussion by shivupa** [BlueBrain/HighFive#707] [Questions] [answered]
Reproducer: https://github.com/shivupa/highfiveh5easyquestion
```
#include <Eigen/Eigen>
#include <highfive/H5Easy.hpp>

int main(int argc, char **argv) {

    Eigen::Matrix<double, 10, 1> col_vec = Eigen::Matrix<double, 10, 1>::Random();
    Eigen::Matrix<double, 1, 10> row_vec = Eigen::Matrix<double, 1, 10>::Random();

    H5Easy::File file("example.h5", H5Easy::File::Overwrite);
    H5Easy::dump(file, "/col_vec", col_vec);
    H5Easy::dump(file, "/row_vec", row_vec);

    return 0;
}
```

Question: When dumping a eigen column vs row vector, I would expect the shape in the hdf5 file to be `(nelem, 1)` or `(1,nelem)`  respectively. Instead when I dump using `H5Easy::dump`, I get the same shape
```
HDF5 "example.h5" {
GROUP "/" {
   DATASET "col_vec" {
      DATATYPE  H5T_IEEE_F64LE
      DATASPACE  SIMPLE { ( 10 ) / ( 10 ) }
      DATA {
      (0): 0.680375, -0.211234, 0.566198, 0.59688, 0.823295, -0.604897,
      (6): -0.329554, 0.536459, -0.444451, 0.10794
      }
   }
   DATASET "row_vec" {
      DATATYPE  H5T_IEEE_F64LE
      DATASPACE  SIMPLE { ( 10 ) / ( 10 ) }
      DATA {
      (0): -0.0452059, 0.257742, -0.270431, 0.0268018, 0.904459, 0.83239,
      (6): 0.271423, 0.434594, -0.716795, 0.213938
      }
   }
}
}
```

Is this expected and if so is there a way to get the `(nelem, 1)` or `(1,nelem)` shapes if desired?

**Comment by 1uc** [comment, marked as answer]
This seems intentional, yes:
https://github.com/BlueBrain/HighFive/blob/master/include/highfive/h5easy_bits/H5Easy_Eigen.hpp#L48
https://github.com/BlueBrain/HighFive/blob/master/include/highfive/h5easy_bits/H5Easy_Eigen.hpp#L89

For row or column vectors you're likely not affected by the bug in core HighFive. Hence, you could try using something like
```
H5Easy::File file("example.h5", H5Easy::File::Overwrite);
file.createDataSet("row_vector", row_vector);
file.createDataSet("column_vector", column_vector);
```
(A `H5Easy::File` is simply a `HighFive::File` it seems.) If I read the output of your reproducer with the modification, it actually gets the order of columns and row right. However, I tend to mix them up. So please check it carefully before using it.

However, you should not start using core HighFive with Eigen matrices or anything genuinely 2D since core HighFive isn't aware that Eigen uses FORTRAN style indexing conventions; while everything else uses C-style.

**Reply by shivupa** [reply]
I see #532 thanks I'll be careful with 2D data. This solved my issue for 1D data. Thank you!
